City-of-Helsinki/drupal-infofinland
# Remove span elements when pasting content to ckeditor

MS-Word adds spans to content which causes CKeditor to show the language tag in editor.

## What was done
Added paste filter to ckeditor which removes the span around pasted content.
Removed permission configs that doesn't exist anymore.
Updated node_type from config, it doesn't exist anymore.

## How to install

* Make sure your instance is up and running on correct branch.
  * `git checkout UHF-X_ckeditor_pastefilter`
  * `make fresh`
* Run `make drush-cr`

## How to test

- Go to edit any content
- Create a span element and paste it in the ckeditor source-view, for example
```
<p>
  <span lang="ar" dir="rtl">this is my test text</span>
</p> 
```

- Leave the source-view.
- In the content-view you should see the test-text surrounded by blue outline
- Copy the whole blue-outlined box 
- Remove the blue outlined box if you wish
- Paste the copied text to the editor 
- The text should appear without the blue outline
- Go check the source again

In the source you should see your test-text surrounded by p-tag. span should have been filtered out.
